### DISCUSSION TITLE
button is being located by Cypress but clicking is not working in automation

### DISCUSSION THREAD

**Discussion by Upsh67** [cypress-io/cypress#29223] [Questions and Help] [answered]
"Save" button is being located by Cypress but clicking is not working in automation, despite manual clicking working fine in the browser, its quite frustrating. I tried several ways (force true, retry, double click, some js $btn click) to click but no luck. Pls help
Thank you

**Comment by samelawrence** [comment, marked as answer]
@Upsh67 You could try using `cy.click( { force: true } );` - or if that doesn't work, you might want to implement the `cypress-real-events` plugin, which I find is more reliable at sending click events that are realistic, like a click with a physical mouse would be. I talk more about how I use this plugin at [this blog post](https://dev.to/samelawrence/essential-cypress-plugins-i-cant-live-without-g6b). I hope it's helpful to you.

**Reply by Upsh67** [reply]
@samelawrence appreciate your response. 
I tried cy.click( { force: true } ) but no luck. I will try cypress-real-events option now, thank you for linking blog post.

**Reply by Upsh67** [reply]
@samelawrence it worked super well. Thank you very much. 
What I did-
1. I installed using this command = npm install cypress-click-events
2. In support folder, in index.js (or command.js) file I imported it = import 'cypress-click-events/support'
Thank u thank you

**Reply by samelawrence** [reply]
@Upsh67 Glad to hear it. I really like this plug-in, and I generally prefer using `cy.realClick()` over `cy.click()` in my tests now.

**Reply by samelawrence** [reply]
@Upsh67 If your issue is resolved, can you mark this discussion as `Answered`? Thanks!

**Comment by vitaliy4us** [comment]
I have the same issue, but unfortunately it can't be solved by installing cypress-real-events plugin. Commands realType() and realClick() are performed and displayed in the log, but actually nothing is typed and nothing is clicked. Also even if the button is clicked manually in Cypress runner it doesn't work. Everything works fine when executing manually in general browser. Could someone advise, please?

**Reply by samelawrence** [reply]
Have you tried using `cy.click( { force: true } );`?

**Reply by JSeifBY** [reply]
@vitaliy4us could you try `.trigger("click")` please ?

**Comment by JSeifBY** [comment]
I had the same issue, clicking on a button that should redirect me to another page (am using a button that calls a javascript function, not a link)

I have tried multiple things, and finally i get it work by simply using `.trigger("click")`, hope that could help other devs.
